### ISSUE TITLE
Why do we use RhsOfName, rather than just Name/HermitName?

### ISSUE THREAD

**Issue by andygill** [ku-fpg/hermit#127]
@xich Is this for better tab completeness? Or can we unify around `Name` in the shell?

**Comment by xich** [comment]
I'm not sure I know specifically where you mean.

There are still places where we use String instead of HermitName that should be fixed. We also have several newtype wrappers for HermitName... those are purely for tab completion. They allow us to distinguish between names in the context vs names that occur in the term, and only tab complete with one or the other, for instance. 

I'm guessing you mean one of those? I tried to keep their use isolated to the externals (as completion is a shell concern), so the actual transformations work on HermitName directly.

**Comment by andygill** [comment]
Yes, this is useful. Given they are semantically the same, we should merge them in the Shell, because there is no tab-complete that can use this. We should also add a note beside the `data` definition.
